torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -83,6 +83,25 @@ std::pair<XLATensor, XLATensor> GetBinaryOperands(const at::Tensor& self,
   return std::pair<XLATensor, XLATensor>(self_tensor, other_tensor);
 }
 
+// input is in format of {N, C, H, W} and the output will be {H, W}.
+std::vector<xla::int64> getOutputSizeWithScale(
+    absl::Span<const xla::int64> input_size,
+    c10::optional<at::ArrayRef<double>> scale_factors,
+    c10::optional<at::IntArrayRef> output_size) {
+  if (!output_size) {
+    XLA_CHECK(scale_factors);
+    XLA_CHECK_EQ(scale_factors->size(), 2);
+    // Calculate the output size from input_shape and scale_factors
+    XLA_CHECK_EQ(input_size.size(), 4);
+    xla::int64 output_h = input_size[2] * (*scale_factors)[0];
+    xla::int64 output_w = input_size[3] * (*scale_factors)[1];
+    return {output_h, output_w};
+  } else {
+    XLA_CHECK(!scale_factors);
```

Comment:
I find it less `indenty` to just follow up in these cases without opening an `else` indentation 😉